src/ui/wxWidgets/SafeCombinationPromptDlg.cpp

```diff
@@ -347,6 +363,33 @@ void SafeCombinationPromptDlg::OnExitClick(wxCommandEvent& WXUNUSED(evt))
 ////@end wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_CANCEL in SafeCombinationPromptDlg.
 }
 
+/*!
+ * wxEVT_COMMAND_CHECKBOX_CLICKED event handler for ID_READONLY
+ */
+
+void SafeCombinationPromptDlg::OnReadonlyClick( wxCommandEvent& event )
+{
+  m_readOnly = event.IsChecked();
+}
+
+void SafeCombinationPromptDlg::UpdateReadOnlyCheckbox(wxCheckBox *checkBox)
+{
```

Comment:
The checkbox object could also be localized via its ID and the function thus made parameterless.

Reply:
This is not needed, as the function is directly called only one time from SafeCombinationPromptDlg::CreateControls() where the reference is known and must search by identifier. The name might be misleading, but is very much the same functionality.